storage/mysql/mysql.go

```diff
@@ -291,6 +300,56 @@ func (s *Storage) integrate(ctx context.Context, tx *sql.Tx, fromSeq uint64, ent
 		}
 	}
 
+	// Add sequenced entries to entry bundles.
+	bundleIndex, entriesInBundle := fromSeq/entryBundleSize, fromSeq%entryBundleSize
+	bundleWriter := &bytes.Buffer{}
+
+	// If the latest bundle is partial, we need to read the data it contains in for our newer, larger, bundle.
+	if entriesInBundle > 0 {
+		row := tx.QueryRowContext(ctx, selectTiledLeavesSQL, bundleIndex)
+		if err := row.Err(); err != nil {
+			return err
+		}
+
+		var partialEntryBundle []byte
+		if err := row.Scan(&partialEntryBundle); err != nil {
+			return fmt.Errorf("row.Scan: %w", err)
+		}
+
+		if _, err := bundleWriter.Write(partialEntryBundle); err != nil {
+			return fmt.Errorf("bundleWriter: %w", err)
+		}
+	}
+
+	// Add new entries to the bundle.
+	for _, e := range sequencedEntries {
+		if _, err := bundleWriter.Write(e.BundleData); err != nil {
+			return fmt.Errorf("bundleWriter.Write: %w", err)
+		}
+		entriesInBundle++
+		fromSeq++
```

Comment:
If you increment `fromSeq` here, will your call to `tb.Integrate` below start building/updating the Merkle tree at the wrong size?